src-theme/src/routes/hud/elements/Speedometer.svelte

```diff
@@ -0,0 +1,120 @@
+<script lang="ts">
+    import {listen} from "../../../integration/ws";
+    import type {ClientPlayerDataEvent} from "../../../integration/events";
+    import type {PlayerData} from "../../../integration/types";
+    import {rgbaToHex, intToRgba} from "../../../integration/util";
+    import {onMount} from "svelte";
+    import {getPlayerData} from "../../../integration/rest";
+    import {fly} from "svelte/transition";
+
+    let playerData: PlayerData | null = null;
+    let speed: number = 0;
+    export let settings: { [name: string]: any };
+    class SpeedDeque {
+        private items: number[] = [];
+        private readonly maxSize: number;
+        constructor(maxSize: number = 20) { // Store last 20 speed samples (1 second at 20 TPS)
+            this.maxSize = maxSize;
+        }
+        addBack(speed: number): void {
+            this.items.push(speed);
+            if (this.items.length > this.maxSize) {
+                this.items.shift();
+            }
+        }
+        getAverage(): number {
+            if (this.items.length === 0) return 0;
+            const sum = this.items.reduce((acc, val) => acc + val, 0);
+            return sum / this.items.length;
+        }
+        getCurrent(): number {
+            return this.items[this.items.length - 1] || 0;
+        }
+        clear(): void {
+            this.items = [];
+        }
+    }
+    const speedHistory = new SpeedDeque(20);
+
+    $: if (playerData?.velocity) {
+        const velocity = playerData.velocity;
+        const totalSpeed = Math.sqrt(velocity.x * velocity.x + velocity.y * velocity.y + velocity.z * velocity.z);
```

Comment:
```suggestion
        const totalSpeed = Math.sqrt(velocity.x * velocity.x + velocity.z * velocity.z);
```
why does the y velocity count? also, why is the theme calculating this?